src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -5104,9 +5144,22 @@ public void visitTypeApply(JCTypeApply tree) {
 
         // Attribute functor part of application and make sure it's a class.
         Type clazztype = chk.checkClassType(tree.clazz.pos(), attribType(tree.clazz, env));
+        Set<JCExpression> valueOKSet = new HashSet<>();
+        List<JCExpression> args = tree.arguments;
+        if (args != null && !args.isEmpty()) {
+            for (Type t : ((ClassType) clazztype.tsym.type).typarams_field) {
+                TypeVar tv = (TypeVar) t;
+                if (tv.universal) {
+                    valueOKSet.add(args.head);
+                }
+                args = args.tail;
+            }
+        }
 
         // Attribute type parameters
-        List<Type> actuals = attribTypes(tree.arguments, env);
+        List<Type> actuals = !allowUniversalTVars ?
```

Comment:
If you are doing this, you might be better off just attributing types one at a time inside the loop above, and avoid using a set to collect the ones which need special treatment?